.ci/set_dependency_version

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+
+import fileinput
+import os
+
+import util
+
+COMPONENT_SPEC_PROJECT_NAME = 'gardener-component-model'
+
+
+def update_project_version(project_name: str, project_version):
+    path_to_requirements_txt = os.path.join(
+            os.path.abspath(util.check_env('MAIN_REPO_DIR')),
+            'requirements.txt',
+        )
+    for line in fileinput.input(path_to_requirements_txt, inplace=True):
+        if line.startswith(project_name):
+            print(f'{project_name}=={project_version}')
+        else:
+            print(line)
+
+
+util.check_env('REPO_DIR')
+
+component_name = util.check_env('DEPENDENCY_NAME')
+component_version = util.check_env('DEPENDENCY_VERSION')
+
+if component_name == 'github.com/gardener/component-spec':
+    update_project_version(COMPONENT_SPEC_PROJECT_NAME, component_version)
+else:
+    raise NotImplementedError(f"Don't know how to upgrade '{component_name}'")
```

Comment:
nit: please always append `=` in format-str-expressions, unless there is a reason against doing so